server/src/main/java/org/elasticsearch/index/IndexMode.java

```diff
@@ -196,8 +216,55 @@ public DocumentDimensions buildDocumentDimensions(IndexSettings settings) {
         public boolean shouldValidateTimestamp() {
             return true;
         }
+
+        /**
+         * For a time series index we just support 'fixed_interval' aggregations.
+         *
+         * @param intervalType the type of interval used by the date histogram
+         */
+        @Override
+        public void validateCalendarIntervalType(
+            IndexSettings indexSettings,
+            DateIntervalWrapper.IntervalTypeEnum intervalType,
+            String valuesSourceDescription,
+            String aggregationName
+        ) {
+            if (indexSettings.getIndexMetadata().isRollupIndex() && DateIntervalWrapper.IntervalTypeEnum.CALENDAR.equals(intervalType)) {
```

Comment:
I *think* we want this test for all time rolled up indices. Not that there are any non-time series ones, but I think the addition of `isRollupIndex` means we don't need these new methods in `IndexMode` any more.